exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/ConvertCountToDirectScanRule.java

```diff
@@ -288,7 +288,10 @@ public void onMatch(RelOptRuleCall call) {
 
           Metadata_V4.ColumnTypeMetadata_v4 columnMetadata = metadataSummary.getColumnTypeInfo(new Metadata_V4.ColumnTypeMetadata_v4.Key(simplePath));
 
-         if (columnMetadata == null || columnMetadata.totalNullCount == Statistic.NO_COLUMN_STATS) {
+          if (columnMetadata == null) {
+            // If the column doesn't exist in the table, row count is set to 0
```

Comment:
Also update the Javadoc for this method by adding this information for non-existing column